CMakeLists.txt

```diff
@@ -11,6 +11,7 @@ eth_policy()
 
 # project name and version should be set after cmake_policy CMP0048
 set(PROJECT_VERSION "0.6.5")
+set(CMAKE_OSX_DEPLOYMENT_TARGET "10.14")
```

Comment:
I'm not sure this would be the right fix... then we'd need to bump the OSX version every now and then

Reply:
This is a minimal target though... wouldn't bumping it be appropriate when the repo becomes incompatible with older os?

Reply:
Hmm yea, that makes sense. I'm wondering *why* it is incompatible though

Reply:
Ok, apparently because of OSX's `libc`, `std::visit` does require 10.14 regardless the compiler version. Weird, but I guess then this is the right fix

Reply:
Can you leave a comment above this `set()` stating it is needed for `std::visit`?